File: python/openai.py
from urllib import request, error
import ssl
import json

headers = {
    'Authorization': 'Bearer token',
    'HOST': 'api.openai.com'
}
url = 'https://52.152.96.252/v1/models'
try:
    opener = request.build_opener()

    opener.addheaders = [headers]
    context = ssl._create_unverified_context()
    req = request.Request(url, headers=headers)
    r = request.urlopen(req, context=context)
    print(r)
except error.HTTPError as e:
    data = e.read()
    # print(data)
    print(json.loads(data))
    

File: nodejs/openai.js

const https = require('https')

const openAi = () => {
    return new Promise((resolve, reject) => {
        let data = ''
        
        const req = https.request({
            port: 443,
            host: '52.152.96.252',
            method: 'GET',
            path: 'https://api.openai.com/v1/models',
            rejectUnauthorized: false,
            timeout: 30,
            headers: {
                Authorization: `Bearer token`
            }
        });

        req.end();
        req.on('response', (res) => {
            res.on('data', chunk => {
                data += chunk
            })
            res.on('end', () => {
                // console.log('end:', data)
                const ret = JSON.parse(data)
                if (ret.error)
                    reject(ret)
                else
                    resolve(ret)
            })
            res.on('error', err => {
                // console.log('err1:', data)
                reject(err)
            })
        });
        req.on('error', err => {
            // console.log('err2:', data)
            reject(err)
        })

    })
}

(async () => {
    try {
            
        const result = await openAi()
        console.log(result)
    } catch (error) {
        console.log('error:', error)
    }
})()